core/store/lib/search/adapter/graphql/gqlQuery.js

```diff
@@ -0,0 +1,78 @@
+export function prepareGraphQlBody (Query) {
+  // @TODO Create graphQl query builder uses gqlQuery.body params
+  // below is a simple demo test products search query
+
+  let query = ``
+  let queryVariables = {}
+  const filters = Query.searchQuery.getAppliedFilters()
+
+  switch (Query.type) {
```

Comment:
**This is important**

We must add the way to extend this logick - I belive there should be a way to extend the graphQL queries support for new entities **without modifying the core**

Would be great if from the Extension or Theme level developer could do sth like:

```js
const factory = new SearchAdapterFactory()
factory.registerEntityType('graphql', 'category', { 
 gql: require('./queries/categories.gql'),
 resultPorcessor: (resp, start, size) =>  {
  const response = {
    items: map(resp.hits.hits, hit => {
      return Object.assign(hit._source, {
        _score: hit._score,
        slug: (hit._source.hasOwnProperty('url_key') && rootStore.state.config.products.useMagentoUrlKeys)
          ? hit._source.url_key
          : (hit._source.hasOwnProperty('name') ? slugify(hit._source.name) + '-' + hit._source.id : '')
      }) // TODO: assign slugs server side
    }), // TODO: add scoring information
    total: resp.hits.total,
    start: start,
    perPage: size,
    aggregations: resp.aggregations
  }

})
```